### PR TITLE
Feature: Make cryptographic algorithms configurable

### PR DESCRIPTION
**Pull request by uliSchuster**
Dear Benny
Thanks for the great work reviving the S/MIME Util library! One of the features you were requesting from the previous maintainer was to make the cryptographic algorithms configurable. I do have the same need for some projects I am working on. Therefore, I added support for this type of configuration, without breaking the existing API. To check that configuration is indeed working, I added several tests. I hope my changes are helpful and fit with your architectural ideas for the library.

Your parent POM did not quite work out for my setup; therefore I removed the upstream dependence; I hope it is not too much of an issue to reintroduce it and make it work for your build pipelines.

### CODE REVIEW

File: src/main/java/org/simplejavamail/utils/mail/smime/SmimeKey.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ private void extractAssociatedAddresses() {
 		try {
 			X509Certificate certificate = getCertificate();
 			if (null != certificate) {
-				Principal principal = certificate.getSubjectDN();
```
Comment (bbottema) [inline src/main/java/org/simplejavamail/utils/mail/smime/SmimeKey.java:92]: Can you elaborate on this change please?

**Comment by bbottema** [comment]
Hi, I finally got some time to look into this. Thank you for your work! I have one review question open before I merge and release, please have a look.

File: src/main/java/org/simplejavamail/utils/mail/smime/SmimeKey.java (hunk 1 above)
Comment (uliSchuster) [reply src/main/java/org/simplejavamail/utils/mail/smime/SmimeKey.java:92]: According to its JavaDocs, `getSubjectDN()` is deprecated:

> Deprecated
> Use getSubjectX500Principal instead. This method returns the subject as an implementation specific Principal object, which should not be relied upon by portable code.

**Comment by uliSchuster** [comment]
Sorry for my late reply, I just returned from vacation.

File: src/main/java/org/simplejavamail/utils/mail/smime/SmimeKey.java (hunk 1 above)
Comment (uliSchuster) [reply src/main/java/org/simplejavamail/utils/mail/smime/SmimeKey.java:92]: Dear Benny
Any news on my PR? We would like to update our dependencies and are, therefore, hoping on a new release. Thanks very much!

**Comment by bbottema** [comment]
Change merged and released in v2.1.0. Thanks!

**Pull request merged**